appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java

```diff
@@ -1264,6 +1266,116 @@ public void DrawLine(int x1, int y1, int x2, int y2) {
+    for (int i = 1; i < points.length; i++) {
+      path.lineTo(points[i][0] * scalingFactor, points[i][1] * scalingFactor);
+    }
+
+    return path;
+  }
+
+  private float[][] parsePointList(YailList pointList) throws IllegalArgumentException {
+    if (pointList == null) {
+      throw new IllegalArgumentException();
+    }
+    float[][] points = new float[pointList.size()][2];
+    int index = 0;
+    YailList pointYailList;
+    for (Object pointObject : pointList.toArray()) {
+      if (pointObject instanceof YailList) {
+        pointYailList = (YailList) pointObject;
+        if (pointYailList.size() == 2) {
+          try {
+            points[index][0] = Float.parseFloat(pointYailList.getString(0));
+            points[index][1] = Float.parseFloat(pointYailList.getString(1));
+            index++;
+          } catch (NullPointerException e) {
+            throw new IllegalArgumentException(e.fillInStackTrace());
+          } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(e.fillInStackTrace());
+          }
+        } else {
+          throw new IllegalArgumentException("length of item YailList("+ index +") is not 2");
+        }
+      } else {
+        throw new IllegalArgumentException("item("+ index +") in YailList is not a YailList");
+      }
+    }
+    return points;
+  }
+
+  /**
+   * Draw an arc on Canvas, by drawing an arc from a specified oval (specified by left, top, right & bottom).
+   * Start angle is 0 when heading to the right, and increase when rotate clockwise.
+   * When useCenter is true, a sector will be drawed instead of an arc.
+   * When fill is true, a filled arc (or sector) will be drawed instead of just an outline.
+   * 
+   * @param left   the left end of the oval
+   * @param top    the top of the oval
+   * @param right  the right end of the oval
+   * @param bottom the bottom of the oval
+   * @param startAngle the start angle of the arc, rightward as 0, clockwise as positive, in degree
+   * @param sweepAngle the sweep angle of the arc, clockwise as positive, in degree
+   * @param useCenter  fill a sector instead of just an arc (when true)
+   * @param fill  true for filled arc (or sector); false for outline only
+   */
+  @SimpleFunction(description = 
+      "Draw an arc on Canvas, by drawing an arc from a specified oval (specified by left, top, right & bottom). " +
+      "Start angle is 0 when heading to the right, and increase when rotate clockwise. " +
+      "When useCenter is true, a sector will be drawed instead of an arc. " +
+      "When fill is true, a filled arc (or sector) will be drawed instead of just an outline.")
+  public void DrawArc(int left, int top, int right, int bottom, 
+      float startAngle, float sweepAngle, boolean useCenter, boolean fill) {
+    float scalingFactor = $form().deviceDensity();
+    Paint p = new Paint(paint);
+    p.setStyle(fill ? Paint.Style.FILL : Paint.Style.STROKE);
+    view.canvas.drawArc(
```

Comment:
This signature of `drawArc` was added in API level 21. Please rewrite this to use the version that is available from API 1 so that it works on versions of Android older than 5.1.